### ISSUE TITLE
[ERROR]: Unable to parse results from queryads.php: (boolean)

### ISSUE THREAD

**Issue by Lukas1818** [pi-hole/docker-pi-hole#1096]
### Versions
PIHOLE_DOCKER_TAG: 2022.04.3

### Platform
- Platform:  Docker at Debian

### Expected behavior
Open web interface without error.

### Actual behavior / bug
display error:
![image](https://user-images.githubusercontent.com/44570204/168351073-30f881fd-a950-4d8d-93e2-a08ab8ceca1e.png)

### Steps to reproduce

open https://dns.lukas1818.de/

### Additional context
My caddy config:
```
dns.lukas1818.de {
	respond /robots.txt 200 {
		body "User-agent: *
Disallow: /"
	}
	@overTimeData {
		expression {query}.contains("overTimeData")
	}
	respond @overTimeData 403

	reverse_proxy localhost:10501 {
		header_down Referrer-Policy "strict-origin-when-cross-origin"
		header_down Strict-Transport-Security "max-age=15768000"
		header_down X-Frame-Options "sameorigin"
	}
}
```

I don not know when it does stop working, but last year it had work without any issue.

**Comment by dschaper** [comment]
If you are Experiencing issues with a Pi-hole install that has non-standard elements (e.g you are using nginx instead of lighttpd, or there is some other aspect of your install that is customised) - please use the [Community Help](https://discourse.pi-hole.net/c/bugs-problems-issues/community-help/36) category.

**Comment by dschaper** [comment]
https://dns.lukas1818.de/admin/

**Comment by warent** [comment]
I'm having this problem with Caddy as well. Can you share how you got it working?

**Comment by warent** [comment]
Oh I see, if you add `/admin` to the URL then it shows up fine. It just doesn't like the root `/` directory

**Comment by IGitYou** [comment]
I have the same issue using the image: pihole/pihole

This is what I am getting now when going straight to the DNS root as @warent stated above. Adding the URL to the Admin directory seems to show fine. What is expected when going to the root is the PiHole image stating that this is blocked (IIRC) and stating that if you were seeking the admin page how to get there. 

I have errors in my startup but I do not think they relate. Admin seems to mostly work fine.

**Comment by nichady** [comment]
I am getting the same issue in 2022.05 but not in 2022.04.2

**Comment by Ashitaka57** [comment]
I have this issue as well on last version. Did someone have found any clue to solve it?

**Comment by stephane-segning** [comment]
Hi, where you able to solve it @Ashitaka57 ?

**Comment by Ashitaka57** [comment]
No, instead I redirected / on /admin on my proxy_pass line.

**Comment by dbaker-rh** [comment]
I have the same issue, running in a kubernetes cluster.

The error originates from here: https://github.com/pi-hole/pi-hole/blob/master/advanced/index.php#L221-L223  (that file is in /var/www/html/pihole/index.php inside the container image).

Looking at the history of that file, I see some changes around the v5.9 timeline but haven't had time to look closer and find what specific change has caused the difference in behaviour.

**Comment by dbaker-rh** [comment]
It turns out the log files are quite informative here.

/var/log/lighttpd/error-pihole.log
...
2022-07-26 15:51:27: mod_fastcgi.c.487) FastCGI-stderr:PHP Warning:  file(https://127.0.0.1:80/admin/scripts/pi-hole/php/queryads.php?domain=pihole.xxx.xxx&amp;bp): failed to open stream: operation failed in /var/www/html/pihole/index.php on line 186

Note the URL starts https:// (the method of the actual request, prior to the ingress decrypting and forwarding) and the port is 80 (the decrypted/forwarded port that's listened on).



So, specifically, the problem is in the new code here: https://github.com/pi-hole/pi-hole/blob/master/advanced/index.php#L168-L175 that was introduced in this commit https://github.com/pi-hole/pi-hole/commit/c9809371abf89efe4f39c25170773eb2a5f39929 and which mis-detects the protocol in play when in the containerized environment.

I've not spent long looking at it, but it feels like the idea behind the patch is incorrect.  It's trying to work out how the end user reached the pihole instance and use that same method to make a call to itself within this webpage.  Instead, it may make more sense to *KNOW* where the local instance is running and access that directly regardless of how the end user reached it.

So, long story short, to make it work a small change in the http/https detection is needed.  I opened https://github.com/pi-hole/pi-hole/pull/4840 with a change that I've tested locally.
